src/_macosx.m

```diff
@@ -178,6 +178,101 @@ static int wait_for_stdin(void)
     return 1;
 }
 
+static PyObject* set_icon(PyObject* ignored, PyObject* path)
+{
+    if (!path)
+    {
+        PyErr_SetString(PyExc_RuntimeError, "No path passed");
+        return NULL;
+    }
+
+    const char* path_string;
+    PyObject* unicode_path = PyObject_Str(path);
+    if (!unicode_path)
+    {
+        return NULL;
+    }
+
+    // API indicates Python owns path_string; we don't need to free it
+    path_string = PyUnicode_AsUTF8(unicode_path);
+    Py_DECREF(unicode_path);
+    if (!path_string)
+    {
+        PyErr_SetString(PyExc_TypeError, "bad argument type, must be "
+                "convertible to UTF-8");
+        return NULL;
+    }
+
+    NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
+
+    NSImage* image;
+    @try
+    {
+        NSString* path_nsstring =
+            [[NSString alloc] initWithUTF8String: path_string];
+
+        image =
+            [[NSImage alloc] initByReferencingFile: path_nsstring];
+    }
+    @catch (NSException *e)
+    {
+
```

Comment:
please cut down a bit on the blank lines :p

Reply:
Sure, I’ll condense it.

Reply:
See latest push.